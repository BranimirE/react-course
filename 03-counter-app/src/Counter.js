import {useState} from 'react';
import PropTypes from 'prop-types';

const Counter = ({start:value = 10}) => {
    const [count, setCount] = useState(value);
    const handleAdd = () => setCount(count + 1);
    const handleReset = () => setCount(value);
    const handleSubstract = () => setCount(count - 1);
    return (
        <>
            <h2>{count}</h2>
            <button onClick={handleSubstract}>-1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleAdd}>+1</button>
        </>
    );
};

/* Counter.propTypes = {
    start: PropTypes.number.isRequired
} */


export default Counter;