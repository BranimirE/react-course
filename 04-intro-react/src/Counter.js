import {useState} from 'react';
import PropTypes from 'prop-types';

const Counter = ({start:value}) => {
    const [count, setCount] = useState(value);
    const handleAdd = () => setCount(count + 1);
    const handleReset = () => setCount(value);
    const handleSubstract = () => setCount(count - 1);
    return (
        <>
            <h2>{count}</h2>
            <button onClick={handleSubstract}>-1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleAdd}>+1</button>
        </>
    );
};

Counter.propTypes = {
    start: PropTypes.number.isRequired
}

// Counter.defaultProps = {
//     start: 5
// }

export default Counter;