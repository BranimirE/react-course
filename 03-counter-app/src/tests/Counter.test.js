import { shallow } from 'enzyme';
import Counter from '../Counter';
describe('Tests in Counter component', () => {
    let wrapper = shallow(<Counter />);

    beforeEach(() => {
        wrapper = shallow(<Counter />);
    });

    test('Should show <Counter /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show 100 as default value', () => {
        const startValue = 100;
        const wrapper = shallow(<Counter start={startValue}/>);
        const defaultStart = wrapper.find('h2').text().trim();
        expect(defaultStart).toBe(startValue + '');
    });

    test('should increase the value using +1 button', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('11')
    });

    test('should decrease the value using -1 button', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('9')
    });

    test('should reset the value when we click on reset button', () => {
        const startValue = 100;
        const wrapper = shallow(<Counter start={startValue}/>);
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        //reset click
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(`${startValue}`);
    });
});
