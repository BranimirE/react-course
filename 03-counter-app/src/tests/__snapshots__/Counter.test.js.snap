// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Counter component Should show <Counter /> correctly 1`] = `
<Fragment>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
